azurerm/resource_arm_automation_bool_variable.go

```diff
@@ -0,0 +1,42 @@
+package azurerm
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceArmAutomationBoolVariable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmAutomationBoolVariableCreateUpdate,
+		Read:   resourceArmAutomationBoolVariableRead,
+		Update: resourceArmAutomationBoolVariableCreateUpdate,
+		Delete: resourceArmAutomationBoolVariableDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: AutomationVariableCommonSchemaFrom(map[string]*schema.Schema{
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"value": {
```

Comment:
As all the resources share this property asie from type you should beable to pass in the type and declare this in the common schema function maybe also pass in a validation function that too

Reply:
I keep this one in the individual resources because the null variable does not have this property.

Reply:
Good call